### ISSUE TITLE
Not all Factorizations have Adjoint (namely QR varients)

### ISSUE THREAD

**Issue by oxinabox** [JuliaLang/julia#38293]
Encountered during https://github.com/JuliaDiff/ChainRules.jl/pull/302
the simpler form is:

I am calling for example:
```julia
Af = factorize(A)
C1 = Af\B1
C2 = Af'\B2
```

Which I thought should be a faster way to do
```julia
C1 = A\B1
C2 = A'\B2
```
but it errors depending on the value of `A`.
.e.g works if `A` is square since `LU` has `Adjoint`
I am not sure if it works if positive definite since then it gets `Cholesky` which may or may not have `Adjoint` defined on it.

But neither `QRPivoted` nor `QRCompactWY` support `adjoint`
I believe fixing this would also require adding a method for `ldiv!` on that adjoint.

Current status:

- `BunchKaufman`
  - [ ] `adjoint`
  - [ ] `\`
- `Cholesky`
  - [ ] `adjoint`
  - [ ] `\`
- `CholeskyPivoted`
  - [ ] `adjoint`
  - [ ] `\`
- `Hessenberg`
  - [x] `adjoint`
  - [ ] `\`
- `LDLt`
  - [ ] `adjoint`
  - [ ] `\`
- `LQ`
  - [x] `adjoint`
  - [ ] `\`
- `LU`
  - [x] `adjoint`
  - [x] `\`
- `LinearAlgebra.QRCompactWY`
  - [ ] `adjoint`
  - [ ] `\`
- `QR`
  - [ ] `adjoint`
  - [ ] `\`
- `QRPivoted`
  - [ ] `adjoint`
  - [ ] `\`
- `SVD`
  - [ ] `adjoint`
  - [ ] `\`

Some factorizations don't support `\` at all
- `Eigen`
- `GeneralizedEigen`
- `Schur`
- `GeneralizedSchur`
- `GeneralizedSVD`

For sparse matrices,`cholesky` and `lu` already works and the sparse QR is tracked in https://github.com/JuliaLang/SuiteSparse.jl/issues/22

**Comment by mschauer** [comment]
Ref https://arxiv.org/pdf/1710.08717.pdf

**Comment by oxinabox** [comment]
> Ref https://arxiv.org/pdf/1710.08717.pdf

I mean adjoint on the QR object (i.e. turn it sidewise),
**not** the abuse of terminology that is the adjoint of the function (i.e. find a linear operator that has the same rate of change, then find the linear operator that has as it's matrix form the adjoint of the first's matrix form)

**Comment by dkarrasch** [comment]
related: #35421

**Comment by andreasnoack** [comment]
Fixed by #40899
